### DISCUSSION TITLE
PID Autotune in the menu

### DISCUSSION THREAD

**Discussion by ottobjorkland** [Jyers/Marlin#343] [Ideas]
## The idea
It would be great if PID was able to be tuned directly from the display menu. It could be placed under the 'Advanced' menu, called 'PID Autotune'. When clicking this button, you will be promted to choose to tune the hotend or the heated bed. After this, you also need to choose a preferred heating temperature. This feature is basically sending `M303 E{'0' for hotend or '-1' for bed} S{Preferred Temperature} U1` to the 3D-printer.

## Requirements
I don't know if this already is added, but it is required that `PIDTEMPBED` is enabled in the firmware to be able to tune the heated bed.

## Why add this feature?
It will make the life so much easier for those who don't know much about sending commands to the printer. There is also a chance that this will fix issues with layer lines. Normally the bed's temperature is fluctuating, making the bed to expand and contract during print, leading to small defects between layers.

_Thank you @Jyers for giving your time into the community!_

**Comment by joeclifford** [comment]
This has been done already and is in the current code: just waiting for a new release with the updated code.

See commit: https://github.com/Jyers/Marlin/commit/b7be23d53b7f89ee466f3613a5c7918a0004ba05

**Reply by joeclifford** [reply]
@ottobjorkland, alternatively you can compile the firmware yourself and get the feature now before the next release.
